src/group_theory/exponent.lean

```diff
@@ -150,11 +169,54 @@ begin
     simp only [true_and, set.mem_univ, finset.coe_univ] at hg,
     exact ne_of_gt (order_of_pos g) hg },
   { intro g,
-    have h : (order_of g) ∣ (finset.univ : finset H).lcm order_of,
+    have h : (order_of g) ∣ (finset.univ : finset G).lcm order_of,
     { apply finset.dvd_lcm,
       exact finset.mem_univ g },
     cases h with m hm,
     rw [hm, pow_mul, pow_order_of_eq_one, one_pow] },
 end
 
+end left_cancel_monoid
+
+section group
+
+variable [group G]
+
+@[to_additive]
+lemma _root_.is_cyclic.exponent_eq_card [is_cyclic G] [fintype G] : exponent G = fintype.card G :=
```

Comment:
The converse should not be too difficult to prove using `is_cyclic_of_order_of_eq_card`, but I didn't try.

Reply:
does this hold for non-abelian groups? i can work out an obvious proof for abelian groups but not put too much thought into non-abelian yet :)

Reply:
You are right, abelian is needed. S3 should be a counterexample for nonabelian groups.